features/lib/step_definitions/scenarios_and_steps.rb

```diff
@@ -109,3 +109,26 @@ def snake_case(name)
 Given('a step definition that looks like this:') do |content|
   write_file("features/step_definitions/steps#{SecureRandom.uuid}.rb", content)
 end
+
+Given('a scenario {string} that fails once, passes and then skips scenario') do |full_name|
+  name = snake_case(full_name)
+
+  create_feature("#{full_name} feature") do
+    create_scenario(full_name) do
+      """
+        Given it fails once, then passes
```

Comment:
So it looks like we still need to generate a step def for this one, right?

We need a way to make it fail once, but then pass on the second attempt?

We can't store state in the World (the scenario context) as we get a fresh instance every time the scenario runs, so I'd be tempted to write a step def something like this:

```
step_definition('/^it fails once, then passes/', 'raise "eek" unless $flag;$flag = true')
```

Do you want to give that a try?

Reply:
Oh wait I see from the commit history you already tried something like that, sorry. Maybe I'm reading the code wrong but it seems like there's no step def for this step at the moment?

Reply:
It is actually. Every scenario has a `Background` statement where we create 3 step definition, one of them is used here.
What i'm starting to realise is that `it fails once, then passes` step def has its own ${name} variable. And when `it fails once, then passes` executes second time it isn't flaky anymore. Still the tests has the same fail reason as the "valid" one. I didn't really want to introduce 3rd flaky scenario e.g. fails 3 times and passes but for now i have no other idea. Any thoughts guys?

Reply:
@mattwynne let me come again:
Yes, there is already built-in step for `it fails once, then passes`. Inside this step definition there is `$#{name}` initialization then the flakiness achieves by  `"$#{name} += 1"` and `"expect($#{name}).to be > 1"`. Since this step always runs as a `Background` refering to it second time gives true here without failing`"expect($#{name}).to be > 1"` and does not simulates flakiness. This is why in order to use "built-in" step `it fails once, then passe` it is required to reset #{name} again, to do that i need to pass the same #{name} to both steps like this:
```
  Background:
    Given a scenario "Fails-once" that fails once, then passes
```
and
```
    Scenario: Flaky scenario gives zero exit code in no-strict-flaky mode if flaky step fails, passes and skips scenario
      And a scenario "Fails-once" that fails once, passes and then skips scenario
```

As for me it confuses and thats why it stopes me to use pre-built flaky scenario.

Reply:
To simplify all things, i would suggest to go for another similar step `it three times, then passes`, unless we decide to leave this test be as part of the suite. cc @mattwynne